api/tests/Unit/UserAuthorizationScopeTest.php

```diff
@@ -0,0 +1,263 @@
+<?php
+
+namespace Tests\Unit;
+
+use App\Models\Pool;
+use App\Models\PoolCandidate;
+use App\Models\Team;
+use App\Models\User;
+use Carbon\Carbon;
+use Database\Seeders\RolePermissionSeeder;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Illuminate\Support\Facades\Auth;
+use Tests\TestCase;
+use Tests\UsesProtectedRequestContext;
+
+use function PHPUnit\Framework\assertEqualsCanonicalizing;
+
+class UserAuthorizationScopeTest extends TestCase
+{
+    use RefreshDatabase;
+    use UsesProtectedRequestContext;
+
+    protected $platformAdmin;
+
+    protected $pool1;
+
+    protected $pool2;
+
+    protected $user1;
+
+    protected $user2;
+
+    protected $candidate1;
+
+    protected $candidate2;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->seed(RolePermissionSeeder::class);
+
+        $this->platformAdmin = User::factory()
+            ->asAdmin()
+            ->create();
+
+        $this->pool1 = Pool::factory()
+            ->for($this->platformAdmin)
+            ->published()
+            ->afterCreating(function (Pool $p) {
+                // normally this happens only when the process operator is added - might want to reconsider that logic
+                $p->team()->create(['name' => 'pool-'.$p->id]);
```

Comment:
If you want to use this for the PoolOperator role, you actually want to use `$p->legacyTeam()`. The `team` relation going forward is used for the new ProcessOperator role. (This will be much less confusing after #10368.) Setting team_id, like you did in api/tests/Unit/PoolCandidateAuthorizationScopeTest.php, still works.

Reply:
Good catch!